docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -7,59 +7,54 @@ There is an [example](https://github.com/kubernetes-sigs/controller-runtime/tree
 in controller-runtime.
 
 It is suggested to use kubebuilder to initialize a project, and then you can
-follow the steps below to add admission webhooks for core types.
+follow the steps below to add admission webhooks for core types. The example shows
+how to set up a mutating webhook following the controller-runtime's webhook builder.
 
-## Implement Your Handler
+## Implement Your Webhook
 
-You need to have your handler implements the
-[admission.Handler](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#Handler)
+You need to have your webhook to implement the
+[admission.CustomDefaulter](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#CustomDefaulter)
 interface.
 
 ```go
-type podAnnotator struct {
-	Client  client.Client
-	decoder *admission.Decoder
-}
+type podAnnotator struct {}
 
-func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admission.Response {
-	pod := &corev1.Pod{}
-	err := a.decoder.Decode(req, pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
+func (a *podAnnotator) Default(ctx context.Context, obj runtime.Object) error {
+    log := logf.FromContext(ctx)
+    pod, ok := obj.(*corev1.Pod)
+    if !ok {
+        return fmt.Errorf("expected a Pod but got a %T", obj)
+    }
 
 	// mutate the fields in pod
+    if pod.Annotations == nil {
+        pod.Annotations = map[string]string{}
+    }
+    pod.Annotations["example-mutating-admission-webhook"] = "foo"
 
-	marshaledPod, err := json.Marshal(pod)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
-}
-```
-
-If you need a client, just pass in the client at struct construction time.
-
-If you add the `InjectDecoder` method for your handler, a decoder will be
-injected for you.
+    log.Info("Annotated pod")
 
-```go
-func (a *podAnnotator) InjectDecoder(d *admission.Decoder) error {
-	a.decoder = d
 	return nil
 }
 ```
 
 **Note**: in order to have controller-gen generate the webhook configuration for
 you, you need to add markers. For example,
-`// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
+`// +kubebuilder:webhook:path=/mutate--v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io` 
+(for core types the `path` is of format `/mutate--<version>-<kind>`).
```

Comment:
Nit: I would _personally_ prefer if we mention the whole path format and mention that the group for core types is an empty string. Maybe something like:
```suggestion
(for webhooks the `path` is of format `/mutate-<group>-<version>-<kind>`. Since this documentation focuses on the core types the group is an empty string).
```

Reply:
+1, I'll make that change, thanks